pkg/kine/drivers/generic/generic.go

```diff
@@ -311,12 +317,21 @@ func getPrefixRange(prefix string) (start, end string) {
 func (d *Generic) query(ctx context.Context, txName, sql string, args ...interface{}) (rows *sql.Rows, err error) {
 	i := uint(0)
 	start := time.Now()
+
+	done, err := d.ac.Admit(ctx)
+	defer done()
+	if err != nil {
+		err = fmt.Errorf("denied: %w", err)
+		return
+	}
```

Comment:
I don't like this `defer done()`, can we instead call it specifically when we want to, right after the SQL transaction is complete?